### PR TITLE
Improve error logging

### PR DESCRIPTION
**Pull request by just-boris**
Issue: the output is misleading

## What I did

When I have a typescript error in my story, I have the following output

```
ERROR in /path/to/project/stories/demo.stories.tsx
./stories/demo.stories.tsx
[tsl] ERROR in /path/to/project/stories/demo.stories.tsx(8,15)
      TS7006: Parameter 'foo' implicitly has an 'any' type.
Child HtmlWebpackCompiler:
                          Asset      Size               Chunks  Chunk Names
    __child-HtmlWebpackPlugin_0  29.4 KiB  HtmlWebpackPlugin_0  HtmlWebpackPlugin_0
    Entrypoint HtmlWebpackPlugin_0 = __child-HtmlWebpackPlugin_0
    [./node_modules/html-webpack-plugin/lib/loader.js!./node_modules/@storybook/core/dist/server/templates/index.ejs] 1.76 KiB {HtmlWebpackPlugin_0} [built]

WARN Broken build, fix the error below.
WARN You may need to refresh the browser.
```

The error message says "fix the error **below**". But the actual error is **above**.

This PR fixes the issue, the output will be now consistently displayed above, whether it comes from webpack or from storybook directly.

## How to test

- Is this testable with Jest or Chromatic screenshots?
- Does this need a new example in the kitchen sink apps?
- Does this need an update to the documentation?

If your answer is yes to any of these, please make sure to include it in your PR.

### CODE REVIEW

File: lib/core/src/server/build-dev.js [hunk 1]
```diff
@@ -337,25 +346,13 @@ export async function buildDevStandalone(options) {
...
-        logger.error(error);
-      }
-
-      if (error.close) {
-        process.exit(1);
-      }
-    }
-    if (options.smokeTest) {
+    
+    if (options.smokeTest || error && error.close) {
```
Comment (shilman) [inline lib/core/src/server/build-dev.js]: when is error.close set?

Comment (just-boris) [reply lib/core/src/server/build-dev.js]: Whenever dev-server has an unrecoverable error

It is set at this line: https://github.com/storybookjs/storybook/blob/next/lib/core/src/server/dev-server.js#L54

Comment (shilman) [reply lib/core/src/server/build-dev.js]: do we want to shut down the server at this point? cc @ndelangen

Comment (ndelangen) [reply lib/core/src/server/build-dev.js]: yeah probably

**Review by ndelangen — APPROVED** [review]

File: lib/core/src/server/build-dev.js (hunk 1 above)
Comment (ndelangen) [reply lib/core/src/server/build-dev.js]: @just-boris do you think you could do a follow up PR that closes the server?

**Pull request merged**